Python/Product/ExternalProfilerDriver/VTuneStackParser.cs

```diff
@@ -143,4 +151,61 @@ static class VTuneStackParser {
         }
 
     }
+
+    class VTuneStackParserForCPP
+    {
+        public static IEnumerable<SampleWithTrace> ParseFromFile(string filename)
+        {
+            if (!File.Exists(filename)) {
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Strings.ErrorMsgPathDoesNotExist, filename));
+            }
+
+            SampleWithTrace current = null;
+            List<PerformanceSample> currentStack = new List<PerformanceSample>();
+
+            uint index = 0;
+            using (var reader = new StreamReader(filename)) {
+                using (CsvReader csvReader = new CsvReader(reader)) {
+
+                    while (csvReader.Read()) {
+                        var row = csvReader.GetRecord<dynamic>();
+                        var dict = row as IDictionary<string, object>;
+
+                            if (dict["Function"] == null || dict["Function"].ToString() == string.Empty) {
+                                if (dict["Function Stack"] == null || dict["Function Stack"].ToString() == string.Empty) {
+                                    current.AddStack(currentStack);
+                                    currentStack = new List<PerformanceSample>();
+                                } else {
+                                    PerformanceSample s = new PerformanceSample(dict["Function Stack"].ToString(), 
+                                                                                dict["CPU Time"].ToString(),
+                                                                                dict["Module"].ToString(),
+                                                                                dict["Function (Full)"].ToString(),
+                                                                                dict["Source File"].ToString(),
+                                                                                dict["Start Address"].ToString());
+                                    currentStack.Add(s);
+                                }
+                            } else {
+                                if (current != null) {
+                                    yield return current;
+                                }
+                                PerformanceSample s = new PerformanceSample(dict["Function"].ToString(), 
```

Comment:
since you are repeating these column names 2 or 3 times, you should create constants.

it could be an inner class:

```
static class Columns {
    const string Function = "Function";
    const string FunctionStack = "Function Stack";
// etc
}
```

Reply:
Please see #4906 instead, it includes many of the revisions suggested here